File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String,Integer> map = new HashMap<>();
        
        for(String part: participant){
            
            if(!map.containsKey(part)){
                map.put(part,1);
            }
            else{
                map.replace(part,map.get(part)+1);
            }
        }
        
        for(String com: completion){
            if(map.get(com)-1 == 0)
                map.remove(com);
            else
               map.replace(com,map.get(com)-1);
        }
        List<String> answer = new ArrayList<>(map.keySet());
        return answer.get(0);
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    di = {name:0 for name in participant}
    for name in participant:
        di[name]+=1
    for name in completion:
        di[name]-=1
    # print(di.items())
    return [name[0] for name in di.items() if name[1]>0][0]